### PR TITLE
[chore] Add test to detect duplicate rule id

### LINKED ISSUE #2135: [tooling]: Add test to check that each rule has a unique `id`
To avoid the mess of #2134 happening again, we should have test to detect duplicate rule `id` (or glossary item key).

### PR DESCRIPTION
**Pull request by Jym77**
Add a test to check that rule ids are unique, avoiding the mess of #2134.

Closes issue(s):

- closes #2135

Need for Call for Review:
This will not require a Call for Review (add test code)

---

### **When creating PR:**

- [X] Make sure you're requesting to **pull a branch** (right side) to the `develop` branch (left side).
- [X] Make sure you do not remove the "How to Review and Approve" section in your pull request description

### **After creating PR:**

- [X] Add yourself (and co-authors) as "Assignees" for PR.
- [X] Add label to indicate if it's a `Rule`, `Definition` or `Chore`.
- [X] [Link the PR to any issue it solves](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue). This will be done automatically by [referencing the issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) at the top of this comment in the indicated place.
- [X] Optionally request feedback from anyone in particular by assigning them as "Reviewers".

### **When merging a PR:**

- [ ] Close any issue that the PR resolves. This will happen automatically upon merging if the PR was correctly linked to the issue, e.g. by referencing the issue at the top of this comment.

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Call for Review period. In case of disagreement, the longer period wins.

### CODE REVIEW

**Review by dan-tripp-siteimprove — APPROVED** [review]
I found some cases where the YAML-style parsing of the frontmatter IDs can lead to unexpected results.  These cases might never happen in the real world.  I'll mention them anyway.

Case # 1: Octal: 077777 and 32767 would be flagged as duplicate IDs by unique-ids.js.  They probably shouldn't be.
Case # 2: Quotes: 123456 and "123456" would not be flagged as duplicate IDs by unique-ids.js.  They probably should be.

This is similar to the problem I found at https://github.com/act-rules/act-rules.github.io/pull/2121#discussion_r1374923716 .  But in this case I don't know how to change the parsing to make the rule frontmatter IDs get parsed as strings rather than numbers.  So if you want to prevent these cases (which might never happen in the real world), maybe the right place to do it is in the frontmatter.js test: assert that (typeof frontmatter.id) == 'string'.

File: _rules/__tests__/unique-ids.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const getMarkdownData = require('../../utils/get-markdown-data')
+const rulesData = getMarkdownData(`./_rules`)
+
+describe('Rule ids verification', () => {
+	const duplicates = rulesData.filter(
+		ruleData => duplicate(rulesData, ruleData, (rule1, rule2) => rule1.frontmatter.id === rule2.frontmatter.id) !== 1
```
Comment (tbostic32) [inline _rules/__tests__/unique-ids.js]: This might be over simplification, but you could do
```js
ruleData => rulesData.findIndex(r => r.frontmatter.id == ruleData.frontmatter.id) >= 0
```
and then not need the duplicate function.

**Review by tbostic32 — APPROVED** [review]

**Review by carlosapaduarte — APPROVED** [review]
